### PR TITLE
Add React samples

### PR DESCRIPTION
**Pull request by compulim**
@billba @danmarshall @eanders-MS @GeekTrainer

Let's add a new React samples.

### CODE REVIEW

**Comment by compulim** [comment]
Also fixing some coding style mentioned in #621.

File: samples/backchannel/index.html [hunk 1]
```diff
@@ -21,74 +21,80 @@
...
+        domain: params['domain'],
+        secret: params['s'],
+        token: params['t'],
+        webSocket: params['webSocket'] && params['webSocket'] === 'true' // defaults to true
+      });
 
-            const changeBackgroundColor = (newColor) => {
-                document.body.style.backgroundColor = newColor;
-            }
+      BotChat.App({ bot, botConnection, user }, document.getElementById('BotChatGoesHere'));
```
Comment (danmarshall) [inline samples/backchannel/index.html]: will this run in all browsers? Some customers are in IE11.

File: samples/backchannel/index.html [hunk 2]
```diff
@@ -21,74 +21,80 @@
...
 
-            const postButtonMessage = () => {
-                botConnection
-                    .postActivity({type: "event", value: "", from: {id: "me" }, name: "buttonClicked"})
-                    .subscribe(id => console.log("success"));
-            }
+      const changeBackgroundColor = newColor => {
```
Comment (danmarshall) [inline samples/backchannel/index.html]: does fat arrow work in IE ?

File: samples/fullwindow/index.html [hunk 3]
```diff
@@ -1,65 +1,66 @@
...
 
-    For ease of testing, several parameters can be set in the query string:
-        * s = Direct Line secret, or
-        * t = Direct Line token (obtained by calling Direct Line's Generate Token)
-        * domain = optionally, the URL of an alternate Direct Line endpoint
-        * webSocket = set to 'true' to use WebSocket to receive messages (currently defaults to false)
-        * userid, username = id (and optionally name) of bot user
-        * botid, botname = id (and optionally name) of bot
+  For ease of testing, several parameters can be set in the query string:
+    * s = Direct Line secret, or2
```
Comment (danmarshall) [inline samples/fullwindow/index.html]: typo

File: samples/backchannel/index.html (hunk 1 above)
Comment (compulim) [inline samples/backchannel/index.html]: Oh right, will check on IE11.

File: samples/backchannel/index.html (hunk 2 above)
Comment (compulim) [inline samples/backchannel/index.html]: No. Will move it back to `function()`.

File: samples/fullwindow/index.html (hunk 3 above)
Comment (compulim) [inline samples/fullwindow/index.html]: Good catch.

File: samples/backchannel/index.html (hunk 2 above)
Comment (compulim) [inline samples/backchannel/index.html]: Noticed `/backchannel` samples right now also using arrow function, will move them to traditional one too.

**Comment by compulim** [comment]
Tested with IE11 except speech.

**Pull request merged**